### PR TITLE
MULE-16491: Pattern matching for error types in error handlers' type attribute

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java [hunk 1]
```diff
@@ -186,6 +185,28 @@ public void onErrorTypeMatchSeveral() throws Exception {
     assertThat(getPayloadAsString(result), is("false apt1 apt2"));
   }
 
+  @Test
+  public void onErrorTypeMatchNameWildcard() throws Exception {
+    Message result = flowRunner("onErrorTypeMatchNameWildcard").withPayload(true).run().getMessage();
+    assertThat(result, is(notNullValue()));
+    assertThat(getPayloadAsString(result), is("true apt1 apt2"));
```
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: There is a`org.mule.functional.junit4.matchers.MessageMatchers#hasPayload` than can be useful.

File: integration/src/test/resources/org/mule/test/integration/exceptions/on-error-continue-use-case-flow.xml [hunk 2]
```diff
@@ -138,6 +138,48 @@
         </error-handler>
     </flow>
 
+    <flow name="onErrorTypeMatchNameWildcard">
+        <choice>
+            <when expression="#[mel:payload]">
```
Comment (afelisatti) [inline integration/src/test/resources/org/mule/test/integration/exceptions/on-error-continue-use-case-flow.xml]: Remove `mel` mentions.

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java [hunk 3]
```diff
@@ -186,6 +185,28 @@ public void onErrorTypeMatchSeveral() throws Exception {
...
+
+    result = flowRunner("onErrorTypeMatchSeveral").withPayload(false).run().getMessage();
```
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: Why are you testing this flow too?

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java [hunk 4]
```diff
@@ -186,6 +185,28 @@ public void onErrorTypeMatchSeveral() throws Exception {
...
+    assertThat(getPayloadAsString(result), is("false apt1 apt2"));
+  }
+
+  @Test
+  public void onErrorTypeMatchNamespaceWildcard() throws Exception {
+    Message result = flowRunner("onErrorTypeMatchNamespaceWildcard").withPayload(true).run().getMessage();
+    assertThat(result, is(notNullValue()));
+    assertThat(getPayloadAsString(result), is("true apt1 apt2"));
+
+    result = flowRunner("onErrorTypeMatchSeveral").withPayload(false).run().getMessage();
```
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: I think you meant to test with false the same flow.

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java (hunk 1 above)
Comment (eze210) [reply integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: hasPayload used

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java (hunk 3 above)
Comment (eze210) [reply integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: It was an error

File: integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java (hunk 4 above)
Comment (eze210) [reply integration/src/test/java/org/mule/test/integration/exceptions/OnErrorContinueTestCase.java]: You're right

**Pull request merged**